### PR TITLE
Eslint react update

### PR DESCRIPTION
**Pull request by isBatak**
Preparation for today's meeting. 
We still need to test this...

### CODE REVIEW

File: eslint-config/package.json [hunk 1]
```diff
@@ -14,14 +14,15 @@
   },
   "homepage": "https://github.com/infinum/js-linters#readme",
   "dependencies": {
-    "eslint-config-airbnb-base": "^12.1.0"
+    "@typescript-eslint/eslint-plugin": "^2.24.0",
```
Comment (kristijanbambir) [inline eslint-config/package.json]: As discussed, let's use ~ instead of ^.

File: eslint-config/package.json [hunk 2]
```diff
@@ -14,14 +14,15 @@
...
+    "@typescript-eslint/parser": "^2.24.0",
+    "eslint-config-prettier": "^6.10.0",
+    "eslint-plugin-prettier": "^3.1.2"
   },
   "devDependencies": {
-    "eslint": "4.x",
-    "eslint-plugin-import": "2.x"
+    "eslint": "^6.x"
```
Comment (kristijanbambir) [inline eslint-config/package.json]: ^ is the same as 6.x, so both are not needed.

File: eslint-config-react/index.js [hunk 3]
```diff
@@ -1,53 +1,28 @@
 module.exports = {
-  plugins: [
-    'react',
-    'jsx-a11y',
-    'emotion',
+  extends: [
+    'plugin:react/recommended',
+    'plugin:jsx-a11y/recommended',
+    '@infinumjs',
```
Comment (kristijanbambir) [inline eslint-config-react/index.js]: As discussed, this should be the first entry.

File: eslint-config-react/index.js [hunk 4]
```diff
@@ -1,53 +1,28 @@
...
-    }],
-    'object-curly-spacing': 0,
+  plugins: ['react-hooks', 'jsx-a11y'],
+  settings: {
+    react: {
+      version: 'detect'
+    },
   },
+  rules: {
+    'react/react-in-jsx-scope': 0,
```
Comment (kristijanbambir) [inline eslint-config-react/index.js]: 'off' would be better, to be consistent with the base config.

File: eslint-config-react/index.js [hunk 5]
```diff
@@ -1,53 +1,28 @@
...
+    'react/prop-types': [
+      'error',
+      {
+        'ignore': [],
```
Comment (kristijanbambir) [inline eslint-config-react/index.js]: Check if this and the following properties can be omitted.

**Comment by isBatak** [comment]
- fixed all the comments 
- checked if the package is working on TPE project
Everything works so we can release the 2.0 version!

**Pull request merged**